packages/database/src/migrations/20200925020630-AddStackedBarGraphsBySchoolTypeLaosSchools-modifies-data.js

```diff
@@ -0,0 +1,215 @@
+    name: 'Students Following MoES Programmes at Home, % of Schools',
+    dataElementToString: {
+      SchCVD017a: { valueOfInterest: 'Yes', displayString: 'TV' },
+      SchCVD017b: { valueOfInterest: 'Yes', displayString: 'Radio' },
+      SchCVD017c: { valueOfInterest: 'Yes', displayString: 'Online (Facebook, YouTube, etc.)' },
+    },
+  },
+];
+
+const createSingleElementSubDataBuilderConfig = (dataElement, schoolType) => ({
+  dataBuilder: 'countByAllDataValues',
+  dataBuilderConfig: {
+    dataElementCodes: [dataElement],
+    convertToPercentage: true,
+    entityAggregation: {
+      dataSourceEntityType: 'school',
+    },
+    dataSourceEntityFilter: {
+      attributes: {
+        type: schoolType,
+      },
+    },
+  },
+});
+
+const createMultipleElementSubDataBuilderConfig = (dataElementToString, schoolType) => ({
+  dataBuilder: 'countMeasureValues',
+  dataBuilderConfig: {
+    measureBuilder: 'groupData',
+    measureBuilderConfig: {
+      groups: {
+        Multiple: {
+          value: '__',
+          operator: 'regex',
+        },
+        // if it doesn't match one of the groups, it will default to the original value
+      },
+      measureBuilder: 'getStringsFromBinaryData',
+      measureBuilderConfig: {
+        entityAggregation: {
+          dataSourceEntityType: 'school',
+        },
+        dataSourceEntityFilter: {
+          attributes: {
+            type: schoolType,
+          },
+        },
+        delimiter: '__', // Because there is a ', ' in 'Online (Facebook, YouTube, etc.)'
+        dataElementToString,
+      },
+    },
+    convertToPercentage: true,
+  },
+});
+
+const createDataBuilderConfig = (dataElement, dataElementToString) => ({
+  dataBuilders: {
+    [PRE_SCHOOL]: {
+      ...(dataElementToString
+        ? createMultipleElementSubDataBuilderConfig(dataElementToString, PRE_SCHOOL)
+        : createSingleElementSubDataBuilderConfig(dataElement, PRE_SCHOOL)),
+      sortOrder: 1,
+    },
+    [PRIMARY_SCHOOL]: {
+      ...(dataElementToString
+        ? createMultipleElementSubDataBuilderConfig(dataElementToString, PRIMARY_SCHOOL)
+        : createSingleElementSubDataBuilderConfig(dataElement, PRIMARY_SCHOOL)),
+      sortOrder: 2,
+    },
+    [SECONDARY_SCHOOL]: {
+      ...(dataElementToString
+        ? createMultipleElementSubDataBuilderConfig(dataElementToString, SECONDARY_SCHOOL)
+        : createSingleElementSubDataBuilderConfig(dataElement, SECONDARY_SCHOOL)),
+      sortOrder: 3,
+    },
+  },
+});
+
+const createViewJson = name => ({
+  name,
+  description:
+    'This report is calculated based on the number of ‘School COVID-19 Response Laos’ survey responses',
+  type: 'chart',
+  chartType: 'bar',
+  valueType: 'percentage',
+  chartConfig: {
+    $all: {
+      stackId: 1,
+    },
+  },
+  renderLegendForOneItem: true,
+  presentationOptions: { hideAverage: true },
+});
+
+const createReport = (id, name, dataElement, dataElementToString) => ({
```

Comment:
I like this 'functional' config rather than `{...BASE_X_CONFIG, variable}` over and over. What do you think?